.github/workflows/build.yaml

```diff
@@ -0,0 +1,33 @@
+name: formula1-race-analysis-build
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install uv
+        uses: astral-sh/setup-uv@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version-file: ".python-version"
+      - name: Activate virtual environment
+        run: |
+          python -m venv .venv
+          source .venv/bin/activate
+      - name: Install the project
+        run: uv sync --all-extras --dev
+      - name: Run Ruff Linter
+        run: .venv/bin/ruff check
```

Comment:
I think all these commands should go through `uv` as well
Take a look - https://docs.astral.sh/uv/guides/tools/

Reply:
You need to make sure that the build job fails if check finds something, I'm almost certain that's the case, but check it out